include/modules.h

```diff
@@ -304,6 +304,16 @@ class CoreExport Module : public classbase, public usecountbase
 	 */
 	virtual void OnUserConnect(LocalUser* user);
 
+	/** Called when before a user quits.
+	 * The details of the exiting user are available to you in the parameter User *user
+	 * This event is only called when the user is fully registered when they quit. To catch
+	 * raw disconnections, use the OnUserDisconnect method.
+	 * @param user The user who is quitting
+	 * @param message The user's quit message (as seen by non-opers)
+	 * @param oper_message The user's quit message (as seen by opers)
+	 */
+	virtual ModResult OnUserPreQuit(User* user, std::string &message);
```

Comment:
This should take a LocalUser not a User.

Also, the & should be against the type not against the variable name.